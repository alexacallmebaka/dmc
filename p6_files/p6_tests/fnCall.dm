a : (x:int) int {
 return 5; 
}

b : (x:int) int { 
  a: int = 5; 
  return a;
}


d : (x:int) int {
 return x; 
}

x : (y:int, f:bool) int {
 return d(5);
}

var1 : int;

main : () void {
  var1 = b(1);
  var1 = x(var1, true);
}