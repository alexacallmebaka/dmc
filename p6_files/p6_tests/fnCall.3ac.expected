[BEGIN GLOBALS]
a
b
d
x
var1
main
[END GLOBALS]
[BEGIN a LOCALS]
x (formal arg of 8 bytes)
[END a LOCALS]
fun_a:      enter a
            getarg 1 [x]
            setret 5
            goto lbl_0
lbl_0:      leave a
[BEGIN b LOCALS]
x (formal arg of 8 bytes)
a (local var of 8 bytes)
[END b LOCALS]
fun_b:      enter b
            getarg 1 [x]
            setret [a]
            goto lbl_1
lbl_1:      leave b
[BEGIN d LOCALS]
x (formal arg of 8 bytes)
[END d LOCALS]
fun_d:      enter d
            getarg 1 [x]
            setret [x]
            goto lbl_2
lbl_2:      leave d
[BEGIN x LOCALS]
y (formal arg of 8 bytes)
f (formal arg of 8 bytes)
tmp0 (tmp var of 8 bytes)
[END x LOCALS]
fun_x:      enter x
            getarg 1 [y]
            getarg 2 [f]
            setarg 1 5
            call d
            getret [tmp0]
            setret [tmp0]
            goto lbl_3
lbl_3:      leave x
[BEGIN main LOCALS]
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            setarg 1 1
            call b
            getret [tmp0]
            [var1] := [tmp0]
            setarg 1 [var1]
            setarg 2 1
            call x
            getret [tmp1]
            [var1] := [tmp1]
lbl_4:      leave main
